apps/site/lib/site_web/views/event_view.ex

```diff
@@ -20,24 +20,12 @@ defmodule SiteWeb.EventView do
     end
   end
 
-  @doc "Returns a list of years with which we can filter events.
-  Defaults to the current datetime if no assigns
-  "
-  @spec year_options(Plug.Conn.t()) :: %Range{:first => Calendar.year(), :last => Calendar.year()}
-  def year_options(%{assigns: %{date: %{year: year}}}) when is_integer(year) do
-    do_year_options(year)
-  end
-
-  def year_options(_) do
-    %{year: year} = Util.now()
-    do_year_options(year)
-  end
+  @doc "Returns a list of years with existing events from conn"
+  @spec years_with_events(Plug.Conn.t()) :: list
+  def years_with_events(%Plug.Conn{assigns: %{years_for_selection: years_for_selection}}),
```

Comment:
I understand this means years where there aren't no events, but my mind keeps wanting to interpret `years_with_events` as some other data structure containing both years and events. I'm not sure what naming would paint a more accurate picture, but just thought I'd flag it here though 😆